pkg/neotest/basic.go

```diff
@@ -297,13 +298,45 @@ func AddSystemFee(bc *core.Blockchain, tx *transaction.Transaction, sysFee int64
 }
 
 // AddNetworkFee adds network fee to the transaction.
-func AddNetworkFee(bc *core.Blockchain, tx *transaction.Transaction, signers ...Signer) {
+func AddNetworkFee(t testing.TB, bc *core.Blockchain, tx *transaction.Transaction, signers ...Signer) {
 	baseFee := bc.GetBaseExecFee()
 	size := io.GetVarSize(tx)
 	for _, sgr := range signers {
-		netFee, sizeDelta := fee.Calculate(baseFee, sgr.Script())
-		tx.NetworkFee += netFee
-		size += sizeDelta
+		if csgr, ok := sgr.(ContractSigner); ok {
+			// from https://github.com/nspcc-dev/neo-go/blob/v0.103.1/pkg/core/blockchain.go#L2814
+
+			require.NoError(t, vm.IsScriptCorrect(csgr.InvocationScript(), nil))
```

Comment:
Do we need to check the script correctness?

Reply:
This is what we do in `Blockchain.InitVerificationContext`. Plus we get it from the user, shouldn't hurt.